src/crypto/bn.js

```diff
@@ -1,130 +1,156 @@
+
+function fromBuffer(buf, opts) {
+  let localbuf;
   if (typeof opts !== 'undefined' && opts.endian === 'little') {
-    buf = reversebuf(buf);
+    localbuf = reversebuf(buf);
+  } else {
+    localbuf = buf;
   }
-  var hex = buf.toString('hex');
-  var bn = new BN(hex, 16);
+  const hex = localbuf.toString('hex');
+  const bn = new BN(hex, 16);
   return bn;
-};
+}
+BN.fromBuffer = fromBuffer;
 
 /**
  * Instantiate a BigNumber from a "signed magnitude buffer"
  * (a buffer where the most significant bit represents the sign (0 = positive, -1 = negative))
  */
-BN.fromSM = function(buf, opts) {
-  var ret;
+function fromSM(buf, opts) {
+  let ret;
+  let localbuf;
   if (buf.length === 0) {
-    return BN.fromBuffer(new Buffer([0]));
+    return BN.fromBuffer(Buffer.alloc([0]));
   }
 
-  var endian = 'big';
+  let endian = 'big';
   if (opts) {
-    endian = opts.endian;
+    endian = opts;
   }
   if (endian === 'little') {
-    buf = reversebuf(buf);
+    localbuf = reversebuf(buf);
+  } else {
+    localbuf = buf;
   }
 
-  if (buf[0] & 0x80) {
-    buf[0] = buf[0] & 0x7f;
-    ret = BN.fromBuffer(buf);
+  if (localbuf[0] && 0x80) {
+    const tmpvar = localbuf[0];
+    /* eslint no-bitwise: ["error", { "allow": ["&"] }] */
+    localbuf[0] = tmpvar & 0x7f;
+    ret = BN.fromBuffer(localbuf);
     ret.neg().copy(ret);
   } else {
-    ret = BN.fromBuffer(buf);
+    ret = BN.fromBuffer(localbuf);
   }
   return ret;
-};
+}
+BN.fromSM = fromSM;
 
 
-BN.prototype.toNumber = function() {
+function toNumber() {
   return parseInt(this.toString(10), 10);
-};
+}
+BN.prototype.toNumber = toNumber;
 
-BN.prototype.toBuffer = function(opts) {
-  var buf, hex;
+function toBuffer(opts) {
+  let buf;
+  let hex;
   if (opts && opts.size) {
     hex = this.toString(16, 2);
-    var natlen = hex.length / 2;
+    const natlen = hex.length / 2;
+    /* eslint-disable */
     buf = new Buffer(hex, 'hex');
+      /* eslint-enable */
 
-    if (natlen === opts.size) {
-      buf = buf;
-    } else if (natlen > opts.size) {
+    if (natlen > opts.size) {
       buf = BN.trim(buf, natlen);
     } else if (natlen < opts.size) {
       buf = BN.pad(buf, natlen, opts.size);
     }
   } else {
     hex = this.toString(16, 2);
-    buf = new Buffer(hex, 'hex');
+    buf = Buffer.alloc(hex, 'hex');
   }
 
   if (typeof opts !== 'undefined' && opts.endian === 'little') {
     buf = reversebuf(buf);
   }
 
   return buf;
-};
+}
+BN.prototype.toBuffer = toBuffer;
 
-BN.prototype.toSMBigEndian = function() {
-  var buf;
+function toSMBigEndian() {
+  let buf;
   if (this.cmp(BN.Zero) === -1) {
     buf = this.neg().toBuffer();
-    if (buf[0] & 0x80) {
-      buf = Buffer.concat([new Buffer([0x80]), buf]);
+    if (buf[0] && 0x80) {
+      buf = Buffer.concat([Buffer.alloc([0x80]), buf]);
     } else {
-      buf[0] = buf[0] | 0x80;
+      const tmpvar = buf[0];
+      /*eslint-disable */
+      buf[0] = tmpvar | 0x80;
+      /* eslint-enable */
```

Comment:
"no-bitwise" rule is disabled in the .eslintrc.json so this shouldn't be needed.